### PR TITLE
SLIM-979_Modify_cucumber_dlms_device_builder_to_work_for_all_properties_given_in_the_scenario

### PR DESCRIPTION
**Pull request by riannefijn**

### CODE REVIEW

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/builders/entities/DlmsDeviceBuilder.java [hunk 1]
```diff
@@ -252,27 +256,45 @@ public DlmsDeviceBuilder withSettings(final Map<String, String> inputSettings) {
...
         }
 
+        /**
+         * For port/logical_id we want to be able to override the default value
+         * to be null to enable testing against a real device.
+         */
+        if (inputSettings.containsKey(PlatformSmartmeteringKeys.LOGICAL_ID)) {
+            if (inputSettings.get(PlatformSmartmeteringKeys.LOGICAL_ID).isEmpty()
+                    || inputSettings.get(PlatformSmartmeteringKeys.LOGICAL_ID).equals("null")) {
+                this.setLogicalId(null);
```
Comment (bjornamkreutz) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/builders/entities/DlmsDeviceBuilder.java]: You can use com.alliander.osgp.cucumber.core.Helpers.getNullOrNonEmptyString. When you have an empty String, you will get a null value.

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/builders/entities/DlmsDeviceBuilder.java [hunk 2]
```diff
@@ -252,27 +256,45 @@ public DlmsDeviceBuilder withSettings(final Map<String, String> inputSettings) {
...
+            } else {
+                this.setLogicalId(Long.parseLong(inputSettings.get(PlatformSmartmeteringKeys.LOGICAL_ID)));
+            }
+        }
+        if (inputSettings.containsKey(PlatformSmartmeteringKeys.PORT)) {
+            if (inputSettings.get(PlatformSmartmeteringKeys.PORT).isEmpty()
+                    || inputSettings.get(PlatformSmartmeteringKeys.PORT).equals("null")) {
```
Comment (bjornamkreutz) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/builders/entities/DlmsDeviceBuilder.java]: You can use com.alliander.osgp.cucumber.core.Helpers.getNullOrNonEmptyString. When you have an empty String, you will get a null value.

**Comment by riannefijn** [comment]
retest this please

**Comment by bjornamkreutz** [comment]
retest this please

**Pull request merged**